Author: Aurelien Jarno <aurel32@debian.org>
Description: Export TI RPC flags/libs in libdap.pc and dap-config
Forwarded: no
Last-Updated: 2021-08-27

--- a/conf/acinclude.m4
+++ b/conf/acinclude.m4
@@ -560,13 +560,15 @@
 
     # First, we need to see if the xdr routines are in libtirpc, libc, librpc,
     # or librpcsvc or libnsl
+    libtirpcreq=
     dap_xdrlib=
 	PKG_PROG_PKG_CONFIG([0.9.0])
 	PKG_CHECK_MODULES([TIRPC], [libtirpc >= 0.2.4],
 		[dap_xdrlib=`echo "${TIRPC_LIBS}" | sed -e 's/^-l//'`
 		AC_SUBST([TIRPC_CFLAGS])
 		AC_SUBST([TIRPC_LIBS])
-		AC_DEFINE([HAVE_LIBTIRPC], [1], [Define to 1 to use libtirpc.])],
+		AC_DEFINE([HAVE_LIBTIRPC], [1], [Define to 1 to use libtirpc.])
+		[libtirpcreq="libtirpc"]],
 		[dap_xdrlib=""]
 	)
 	AS_IF(
@@ -574,6 +576,7 @@
     	[AC_SEARCH_LIBS([xdr_void],[c rpc nsl rpcsvc],[
     	  dap_xdrlib=`echo $ac_res|sed -e 's/^-l//'`],[
     	  AC_MSG_WARN(Cannot locate library containing xdr functions.)])])
+    AC_SUBST([libtirpcreq])
 
     # Added for autoconf 2.59 which appears to not use/set $ac_res. jhrg
     if test -z "$dap_xdrlib" ; then dap_xdrlib=c; fi
--- a/dap-config.in
+++ b/dap-config.in
@@ -56,7 +56,7 @@
 	;;
 
     --cflags)
-	echo "-I${includedir}/libdap @XML2_CFLAGS@ @CURL_CFLAGS@"
+	echo "-I${includedir}/libdap @XML2_CFLAGS@ @CURL_CFLAGS@ @TIRPC_CFLAGS@"
 	;;
 
     --libs)
--- a/libdap.pc.in
+++ b/libdap.pc.in
@@ -10,6 +10,7 @@
 Version: @VERSION@
 Libs: -L${libdir} -ldap
 Libs.private:  @xmlprivatelibs@ @PTHREAD_LIBS@
+Requires: @libtirpcreq@
 Requires.private: @xmlprivatereq@
 Cflags: -I${includedir}/libdap
 
